pkgs/development/compilers/eunomia-bpf/eunomia-cc/default.nix

```diff
@@ -0,0 +1,96 @@
+{ lib
+, stdenv
+, makeWrapper
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, llvmPackages_16
+, elfutils
+, zlib
+}:
+let
+  bpftool = stdenv.mkDerivation {
+    pname = "bpftool";
+    version = "unstable-2023-03-11";
+    src = fetchFromGitHub {
+      owner = "eunomia-bpf";
+      repo = "bpftool";
+      rev = "05940344f5db18d0cb1bc1c42e628f132bc93123";
+      hash = "sha256-g2gjixfuGwVnFlqCMGLWVPbtKOSpQI+vZwIZciXFPTc=";
+      fetchSubmodules = true;
+    };
+
+    buildInputs = [ llvmPackages_16.clang llvmPackages_16.libllvm.dev elfutils zlib ];
+
+    buildPhase = ''
+      make -C src
+    '';
+
+    installPhase = ''
+      # dir structure compatible with `build.rs` from upstream
+      mkdir -p $out/src/libbpf
+      cp -r src/libbpf/include $out/src/libbpf
+      cp src/bpftool $out/src
+    '';
+  };
+
+  vmlinux-headers = fetchFromGitHub {
+    owner = "eunomia-bpf";
+    repo = "vmlinux";
+    rev = "933f83becb45f5586ed5fd089e60d382aeefb409";
+    hash = "sha256-CVEmKkzdFNLKCbcbeSIoM5QjYVLQglpz6gy7+ZFPgCY=";
+  };
+
+in
+rustPlatform.buildRustPackage rec {
+
+  pname = "ecc";
+  version = "1.0.11";
+
+  src = fetchFromGitHub {
+    owner = "eunomia-bpf";
+    repo = "eunomia-bpf";
+    rev = "v${version}";
+    hash = "sha256-UiwS+osyC3gtbQH0bWNsx1p3xYr993/FAZ5d5NKnaBM=";
+  };
+
+  cargoRoot = "compiler/cmd";
+
+  cargoLock.lockFile = ./Cargo.lock;
+
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+  ] ++ (with rustPlatform; [
+    bindgenHook
+  ]);
+
+  buildInputs = [
+    elfutils
+    zlib
+  ];
+
+  preBuild = ''
+    export SANDBOX=1
+    export OUT_DIR=$(pwd)
+    export VMLINUX_DIR=${vmlinux-headers}
+    export BPFTOOL_DIR=${bpftool}
+    cd ${cargoRoot}
```

Comment:
Are you sure this is needed? I thought `cargoRoot` is a known attribute used by `buildRustPackage`.

Reply:
> Are you sure this is needed? I thought `cargoRoot` is a known attribute used by `buildRustPackage`.

It's under a [monorepo](https://github.com/eunomia-bpf/eunomia-bpf/tree/master/compiler) so it indeed needs to specify `cargoRoot`.

Reply:
> > Are you sure this is needed? I thought `cargoRoot` is a known attribute used by `buildRustPackage`.
> 
> It's under a [monorepo](https://github.com/eunomia-bpf/eunomia-bpf/tree/master/compiler) so it indeed needs to specify `cargoRoot`.

Solved by setting `sourceRoot`.